ios/Plugin/CapacitorUpdater.swift

```diff
@@ -673,20 +677,33 @@ extension CustomError: LocalizedError {
     }
 
 
-    private func saveDownloadProgress(_ bytes: Int64) {
+    private func saveDownloadInfo(_ version: String) {
         do {
-            try "\(bytes)".write(to: progressPath, atomically: true, encoding: .utf8)
+            try "\(version)".write(to: updateInfo, atomically: true, encoding: .utf8)
         } catch {
             print("Failed to save progress: \(error)")
         }
     }
-
+    private func getLocalUpdateVersion() -> String { //Return the version that was tried to be downloaded on last download attempt
```

Comment:
Hello, this is a good step forward, but I think that this function could be simplfied

```swift
UserDefaults.standard.string(forKey: "PROGRESSIVE_DOWNLOAD_LAST_VERSION")
 ```
 
 You would set it with
 ```swift
UserDefaults.standard.set("1.0.0", forKey: PROGRESSIVE_DOWNLOAD_LAST_VERSION)
UserDefaults.standard.synchronize()
```

A similar thing can be done in android

Reply:
Hi 👋 ,

The issue with UserDefaults is that it saves data to the disk at regular intervals, not immediately. This means that if the app crashes or is terminated before UserDefaults has a chance to save the data, the app will not recognize any changes made since the last save, and the download will start from 0 again.

The .synchronize method you are proposing [is not recommended to be used by Apple](https://developer.apple.com/documentation/foundation/userdefaults/1414005-synchronize) and is likely to be deprecated in the future, given these concerns, it might not be prudent to rely on this approach.

So it depends if we can neglect this aspect or not, if not, I believe my method might offer a safer alternative !

Reply:
Hi, thank you for brining this to my attention. You are right, the second file is a good idea